### DISCUSSION TITLE
Is pod locked to node?

### DISCUSSION THREAD

**Discussion by Danielss89** [topolvm/topolvm#586] [Q&A] [answered]
When using topolvm and doing a pvc for a pod, will that pod then be locked to the node where the pv is created? Or can the pod be created on node A with volume on node A and then the pod deleted and created on node B but still using volume on node A?

**Comment by peng225** [comment, marked as answer]
Hi Danielss89.

> When using topolvm and doing a pvc for a pod, will that pod then be locked to the node where the pv is created? 

Yes, it will.
When the pod is deleted and then regenerated by something like Deployment controller, it is allocated to the original node again.

> Or can the pod be created on node A with volume on node A and then the pod deleted and created on node B but still using volume on node A?

This kind of things does not happen as for TopoLVM.

**Comment by jar349** [comment]
@peng225 What should I do when I need to drain node A so that I can reboot it?

**Reply by peng225** [reply]
@jar349 While the node is not available, pods that uses TopoLVM volumes on that node cannot be scheduled and stay pending.

The availability in the event of node down should be managed by the application which is using TopoLVM.
